File: server/src/main/java/edu/rmit/highlandmimic/model/Tag.java
package edu.rmit.highlandmimic.model;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder(toBuilder = true)
@Document("tags")
public class Tag {

    @Id
    private String tagId;

    private String tagName;
    private String tagDescription;

    @Builder.ObtainVia(method = "generateRandomColorCode")
    private String tagColorCode;

    public static String generateRandomColorCode() {
        return String.format("#%06X", (int) (Math.random() * 0xFFFFFF));
    }
}

File: client/lib/src/data/models/tag.dart
import 'package:coffee/src/data/remote/response/tag/tag_response.dart';

class Tag {
  final String tagId;
  final String? tagName;
  final String? tagDescription;
  final String? tagColorCode;

  Tag({
    required this.tagId,
    this.tagName,
    this.tagDescription,
    this.tagColorCode,
  });

  factory Tag.fromTagResponse(TagResponse tagResponse) {
    return Tag(
      tagId: tagResponse.tagId,
      tagColorCode: tagResponse.tagColorCode,
      tagDescription: tagResponse.tagDescription,
      tagName: tagResponse.tagColorCode,
    );
  }
}
